lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/richhawkins/CsvDataLoader.java

```diff
@@ -0,0 +1,121 @@
+package com.codedifferently.lesson12.factory.richhawkins;
+
+import com.codedifferently.lesson12.factory.LibraryCsvDataLoader;
+import com.codedifferently.lesson12.models.CheckoutModel;
+import com.codedifferently.lesson12.models.LibraryDataModel;
+import com.codedifferently.lesson12.models.LibraryGuestModel;
+import com.codedifferently.lesson12.models.MediaItemModel;
+import com.opencsv.CSVReader;
+import com.opencsv.exceptions.CsvException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CsvDataLoader implements LibraryCsvDataLoader {
+  @Override
+  public LibraryDataModel loadData() {
+    var model = new LibraryDataModel();
+    model.mediaItems =
+        readMediaItems(
+            this.getClass().getClassLoader().getResource("csv/media_items.csv").getFile());
```

Comment:
Oh I know you can clean this up...

Reply:
Thanks for the push, code is much cleaner now.